Track code coverage for builds.

maciejmrowiec opened the pull request:
Integrate with codecov and report coverage.

Changelog: none

Signed-off-by: Maciej Mrowiec <mrowiec.maciej@gmail.com>

@jimis early xmas gift ;)

maciejmrowiec commented:
Considering adding exclude for .*_test.c files, currently I see they are included in the coverage.

jimis commented:
Thanks @maciejmrowiec, awesome gift! Consider excluding everything under the `tests/` directory.

jimis commented on travis-scripts/script.sh:
```diff
@@ -47,12 +48,14 @@ chmod -R go-w .
 if [ "$JOB_TYPE" = acceptance_tests_common ]
 then
     ./testall --tests=common
+    bash <(curl -s https://codecov.io/bash) -F acceptance_tests ;
```
`acceptance_tests_common`

maciejmrowiec replied:
They are called the same so it can merge coverage for distributed acceptance test executions to get same result as those whole suite was executed in single go.

jimis commented on travis-scripts/script.sh:
```diff
@@ -47,12 +48,14 @@ chmod -R go-w .
 if [ "$JOB_TYPE" = acceptance_tests_unsafe_serial_network_etc ]
 then
     ./testall --gainroot=sudo --tests=timed,slow,errorexit,libxml2,libcurl,serial,network,unsafe
+    bash <(curl -s https://codecov.io/bash) -F acceptance_tests ;
```
`acceptance_tests_rest`

maciejmrowiec commented:
tricky to do so because all gcov file get dumped flat in main dir after running tests, added some regexes to exclude test.c.gov files (most of unit tests follow this) and in addition -g option that says it allows you to exclude dir (not sure if have effect, setting anyways).  Added commit and lets see the how the next report will look.

jimis commented:
> They are called the same so it can merge coverage for distributed
> acceptance test executions to get same result as the whole suite was
> executed in single go.
>
I think it does not have the effect you want. It merges coverage reports
even with different names. In fact, I think that using the same name is
having opposite effect, and *overwrites* the coverage reports, since
cf-serverd coverage is missing.

Try with different names, and lets see how it goes.

maciejmrowiec commented:
It supposed to merge, we used it heavily in mender.
Exclude did not work as well as I'd like. Added commit with config to the service configuration instead tweaking coverage reports send to the service.

maciejmrowiec commented:
Closing this PR, want to try to open a new one with squashed commits and make it run clean instead overwriting results every time I add commits.
